File: scripts/recolor_logo.py
from PIL import Image
import os

input_path = r'c:\Users\Hp\Documents\Github\Tera\public\images\TERA_LOGO_ONLY.png'
output_path = r'c:\Users\Hp\Documents\Github\Tera\public\images\TERA_LOGO_ONLY1.png'

try:
    img = Image.open(input_path).convert("RGBA")
    datas = img.getdata()
    
    new_data = []
    for item in datas:
        # Check if the pixel is not fully transparent
        if item[3] > 0:
            # Change visible pixels to black, preserving alpha
            new_data.append((0, 0, 0, item[3]))
        else:
            new_data.append(item)
            
    img.putdata(new_data)
    img.save(output_path, "PNG")
    print(f"Successfully created {output_path}")
    
except Exception as e:
    print(f"Error: {e}")

File: scripts/recolor_logo.ts
import jimp from 'jimp';

const inputPath = 'c:\\Users\\Hp\\Documents\\Github\\Tera\\public\\images\\TERA_LOGO_ONLY.png';
const outputPath = 'c:\\Users\\Hp\\Documents\\Github\\Tera\\public\\images\\TERA_LOGO_ONLY1.png';

async function recolorLogo() {
    try {
        const img = await jimp.read(inputPath);
        img.scan(0, 0, img.bitmap.width, img.bitmap.height, function (x: number, y: number, idx: number) {
            const alpha = this.bitmap.data[idx + 3];
            // Check if the pixel is not fully transparent
            if (alpha > 0) {
                // Change visible pixels to black, preserving alpha
                this.bitmap.data[idx] = 0;     // Red
                this.bitmap.data[idx + 1] = 0; // Green
                this.bitmap.data[idx + 2] = 0; // Blue
            }
        });

        await img.writeAsync(outputPath);
        console.log(`Successfully created ${outputPath}`);
    } catch (err) {
        console.error(`Error: ${err}`);
    }
}

recolorLogo();
